scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1908,6 +1909,15 @@ class Router(formatOps: FormatOps) {
             Split(Newline, 1).withIndent(2, lastToken, After)
           )
         }
+      // type bounds in type definitions and declarations
+      case FormatToken(_: T.Subtype, _, _)
+          if leftOwner.is[Type.Bounds] &&
+            leftOwner.parent.exists(t => t.is[Defn.Type] || t.is[Decl.Type]) =>
+        val lastToken = leftOwner.parent.get.tokens.last
+        Seq(
+          Split(Space, 0),
```

Comment:
should there be a single-line policy here? not sure, just asking, really.

Reply:
Not sure how that works. If we specify the single line policy, how is it different from the current behaviour. Or is that policy for everything after `<:`?

Reply:
single-line policy would mean no breaks after space. a penalized-break policy means breaks are allowed but a weight is added to each one. there are plenty of examples of both.

Reply:
That makes sense, I added `withSingleLineNoOptimal`